idaes/models_extra/gas_distribution/parsers/tests/test_networkdata_model.py

```diff
@@ -0,0 +1,140 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Tests the pyomo model created by the NetworkData parser
+"""
+
+import pytest
+import os
+import unittest
+from pyomo.common.fileutils import this_file_dir
+from urllib.request import urlopen
+from zipfile import ZipFile
+from io import BytesIO
+from socket import timeout
+from idaes.models_extra.gas_distribution.parsers.gaslib_parser import parse_gaslib_network
+
+pytest.importorskip("plotly", reason="plotly not available")
+
+try:
+    data = urlopen("https://gaslib.zib.de/download/testData/GasLib-11.zip", timeout=3)
+    zipfile = ZipFile(BytesIO(data.read()))
+    files = [zipfile.open(file) for file in zipfile.namelist()]
+    netfile = files[1]
+    scnfile = files[2]
+    net = parse_gaslib_network(netfile, scnfile)
+    m = net.build_model(with_bounds=False)
+
+except timeout:
+    raise unittest.SkipTest('An internet connection is required to test gaslib parsing tools')
+
+@pytest.mark.unit
```

Comment:
Since these tests rely on data from an external source, they should probably be marked with `pytest.mark.integration` rather than unit.